pyrealm/core/solar.py

```diff
@@ -1,26 +1,604 @@
-"""The ``solar`` submodule provides functions and classes to calculate daily solar
-radiation fluxes and other radiative values.
+"""The :mod:`~pyrealm.core.solar` submodule provides functions to calculate
+photosynthetic photon flux density (ppfd), daily solar radiation fluxes and other
+radiative values.
 """  # noqa: D205
 
 import numpy as np
 from numpy.typing import NDArray
 
 from pyrealm.constants import CoreConst
+from pyrealm.core.calendar import LocationDateTime
+from pyrealm.core.utilities import check_input_shapes
 
-# from pyrealm.splash.const import (
-#     kA,
-#     kalb_sw,
-#     kalb_vis,
-#     kb,
-#     kc,
-#     kd,
-#     ke,
-#     keps,
-#     kfFEC,
-#     kGsc,
-#     komega,
-#     pir,
-# )
+
+def calc_distance_factor(nu: NDArray, k_e: float) -> NDArray:
+    r"""Calculates distance factor.
+
+    This function calculates the distance factor :math:`dr` using the method of \
+    :cite:t:`berger:1993a`.
+
+    .. math::
+
+        dr = \left( 1.0 \mathbin{/}
+               \left(\frac{1.0 - k_{e}^2}
+                          {1.0 + k_{e} \cos\left(\nu \cdot \pi \mathbin{/}
+                          180)\right)}
+               \right)
+             \right)^2
+
+    Args:
+        nu: Heliocentric true anomaly (degrees)
+        k_e: Solar eccentricity
+
+    Returns:
+        An array of distance factors
+    """
+
+    dr = (1.0 / ((1.0 - k_e**2) / (1.0 + k_e * np.cos(np.deg2rad(nu))))) ** 2
+
+    return dr
+
+
+def calc_declination_angle_delta(
+    lambda_: NDArray, k_eps: float, k_pir: float
+) -> NDArray:
+    r"""Calculates declination angle delta.
+
+    This function calculates the solar declination angle delta using the method of \
+    :cite:t:`Woolf:1968`.
+
+    .. math::
+
+        \delta = \frac{\arcsin(\sin(\deg2rad(\lambda_))
+            \cdot \sin(\deg2rad(k_{eps})))}{k_{pir}}
+
+    Args:
+        lambda_: heliocentric longitude
+        k_eps: Solar obliquity
+        k_pir: conversion factor from radians to degrees
+
+    Returns:
+        An array of solar declination angle delta
+    """
+
+    delta = np.arcsin(np.sin(np.deg2rad(lambda_)) * np.sin(np.deg2rad(k_eps))) / k_pir
+
+    return delta
+
+
+def calc_lat_delta_intermediates(
+    delta: NDArray, latitude: NDArray
+) -> tuple[NDArray, NDArray]:
+    r"""Calculates intermediate values for use in solar radiation calcs.
+
+    This function calculates :math:`ru` and :math:`rv` which are dimensionless \
+    intermediate values calculated from the solar declination angle delta and the \
+    observation latitude.
```

Comment:
This module seems to be using `\` to shown line breaks in docstrings, as in this example selection.  That isn't / shouldn't be needed.

Reply:
I think I started doing this after you mentioned something a while back about doing line character limits properly. And I think I read somewhere else that it was necessary/useful. I may also just be confused! Happy to remove them just now!